common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ShuffleClient.java

```diff
@@ -42,5 +42,13 @@ public abstract void fetchBlocks(
       String execId,
       String[] blockIds,
       BlockFetchingListener listener,
-      File[] shuffleFiles);
+      boolean toDisk,
+      TmpFileCreater tmpFileCreater);
+
+  /**
+   * An interface to provide approach to create tmp file.
+   */
+  public interface TmpFileCreater {
```

Comment:
shall we move it to a separated file? We may use it outside of `ShuffleClient` in the future.